source/common/http/conn_manager_impl.cc

```diff
@@ -335,13 +335,11 @@ ConnectionManagerImpl::ActiveStream::~ActiveStream() {
     log_handler->log(request_headers_.get(), response_headers_.get(), request_info_);
   }
 
-  if (active_span_) {
-    if (request_info_.healthCheck()) {
-      connection_manager_.config_.tracingStats().health_check_.inc();
-    } else {
-      Tracing::HttpTracerUtility::finalizeSpan(*active_span_, *request_headers_, request_info_,
-                                               *this);
-    }
+  if (request_info_.healthCheck()) {
+    connection_manager_.config_.tracingStats().health_check_.inc();
+  } else {
+    Tracing::HttpConnManFinalizerImpl finalizer{*request_headers_, request_info_, *this};
```

Comment:
I think @RomanDzhabarov has a fair point - all the really expensive stuff is in the actual finalize() call, which gets skipped entirely in the case of a NullSpan (I for some reason was thinking for a moment that more of it was in the constructor, but now recall I deliberately pushed it all into the finalize call for that reason).

I'm leaning slightly towards sticking with the unnecessary instantiation for now, since it keeps things a little tidier. No very strong feelings about it though.